pkg/cache/resource.go

```diff
@@ -46,18 +46,38 @@ const (
 	AnyType = ""
 )
 
-var (
-	// ResponseTypes are supported response types.
-	ResponseTypes = []string{
-		EndpointType,
-		ClusterType,
-		RouteType,
-		ListenerType,
-		SecretType,
-		RuntimeType,
-	}
+// ResponseType enumeration of supported response types
+type ResponseType int
+
+const (
+	Endpoint ResponseType = iota
+	Cluster
+	Route
+	Listener
+	Secret
+	Runtime
+	UnknownType // token to count the total number of supported types
```

Comment:
nit: The `UnknownType` looks a little strange to represent the `total number of supported types`, especially when looking in the `type Snapshot struct` definition below.

Reply:
Yeah... It's also used as a hole for unknowns though. So not sure if it's better to call a hole TotalNumberTypes or the number UnknownType.